### PR TITLE
chore(telemetry): track service naming configurations

### PR DESCRIPTION
**Pull request by wconti27**
This PR adds additional telemetry configuration to the telemetry writer, the new configurations are:
- `DD_TRACE_PEER_SERVICE_MAPPING`
- `DD_SERVICE_MAPPING`
- trace span attribute schema version
- trace remove integration service names enabled
- trace peer service defaults enabled

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/internal/telemetry/constants.py [hunk 1]
```diff
@@ -38,3 +38,7 @@
 TELEMETRY_PROPAGATION_STYLE_EXTRACT = "DD_TRACE_PROPAGATION_STYLE_EXTRACT"
 TELEMETRY_TRACE_SAMPLING_RATE = "DD_TRACE_SAMPLE_RATE"
 TELEMETRY_TRACE_SAMPLING_LIMIT = "DD_TRACE_RATE_LIMIT"
+TELEMETRY_TRACE_SPAN_ATTRIBUTE_SCHEMA = "DD_TRACE_SPAN_ATTRIBUTE_SCHEMA"
+TELEMETRY_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED = "DD_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED"
+TELEMETRY_TRACE_PEER_SERVICE_DEFAULTS_ENABLED = "DD_TRACE_PEER_SERVICE_DEFAULTS_ENABLED"
+TELEMETRY_TRACE_PEER_SERVICE_MAPPING = "DD_TRACE_PEER_SERVICE_MAPPING"
```
Comment (mabdinur) [inline ddtrace/internal/telemetry/constants.py:44]: Should we also track `DD_SERVICE_MAPPING`?

File: ddtrace/internal/telemetry/writer.py [hunk 2]
```diff
@@ -318,6 +324,14 @@ def _app_started_event(self):
...
                 (TELEMETRY_SPAN_SAMPLING_RULES, config._sampling_rules, "unknown"),
                 (TELEMETRY_SPAN_SAMPLING_RULES_FILE, config._sampling_rules_file, "unknown"),
+                (TELEMETRY_TRACE_SPAN_ATTRIBUTE_SCHEMA, SCHEMA_VERSION, "unknown"),
+                (TELEMETRY_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED, _remove_client_service_names, "unknown"),
+                (
+                    TELEMETRY_TRACE_PEER_SERVICE_DEFAULTS_ENABLED,
+                    os.getenv("DD_TRACE_PEER_SERVICE_DEFAULTS_ENABLED", default=False),
+                    "unknown"
+                ),
+                (TELEMETRY_TRACE_PEER_SERVICE_MAPPING, os.getenv("DD_TRACE_PEER_SERVICE_MAPPING", default=""), "unknown"),
```
Comment (mabdinur) [inline ddtrace/internal/telemetry/writer.py:334]: We should avoid reading environment variables here. We should try to use the existing pattern of storing environment variables in a config object (via envier or ddtrace.config) and then reading the configuration object here.

File: ddtrace/internal/telemetry/constants.py (hunk 1 above)
Comment (wconti27) [reply ddtrace/internal/telemetry/constants.py:44]: yup!

**Review by mabdinur — APPROVED** [review]
LGTM. Can you update the PR description. We should add a high level description of the change and the motivation (what would we like to track with this change).

File: ddtrace/settings/peer_service.py [hunk 3]
```diff
@@ -16,6 +16,7 @@ class PeerServiceConfig(object):
     def __init__(self, set_defaults_enabled=None, peer_service_mapping=None):
```
Comment (mabdinur) [inline ddtrace/settings/peer_service.py:16]: TODO: Migrate `PeerServiceConfig` to envier

**Review by mabdinur — DISMISSED** [review]

**Review by mabdinur — APPROVED** [review]

**Review by emmettbutler — APPROVED** [review]

**Pull request merged**